Marlin/Marlin_main.cpp

```diff
@@ -5039,17 +5039,17 @@ void controllerFan()
   {
     lastMotorCheck = millis();
 
-    if(!READ(X_ENABLE_PIN) || !READ(Y_ENABLE_PIN) || !READ(Z_ENABLE_PIN) || (soft_pwm_bed > 0)
+    if((!READ(X_ENABLE_PIN) ^ !(X_ENABLE_ON)) || (!READ(Y_ENABLE_PIN) ^ !(Y_ENABLE_ON)) || (!READ(Z_ENABLE_PIN) ^ !(Z_ENABLE_ON)) || (soft_pwm_bed > 0)
     #if EXTRUDERS > 2
-       || !READ(E2_ENABLE_PIN)
+       || (!READ(E2_ENABLE_PIN) ^ !(E_ENABLE_ON))
     #endif
     #if EXTRUDER > 1
       #if defined(X2_ENABLE_PIN) && X2_ENABLE_PIN > -1
-       || !READ(X2_ENABLE_PIN)
+       || (!READ(X2_ENABLE_PIN) ^ !(X_ENABLE_ON))
```

Comment:
What does READ return? Is it 0 and 1 or is it 0 and non-zero? Also, you could use != instead of ^ if you prefer.

    !0 ^ !0 = 0
    !0 ^ !1 = 1
    !1 ^ !0 = 1
    !1 ^ !1 = 0